src/components/interaction-designer/block-editors/choices/TextSynonymsEditor.vue

```diff
@@ -1,13 +1,14 @@
 <template>
   <div class="text-synonyms-editor">
     <validation-message
-      v-for="({ test_expression, globalIndex }, synonymIndex) in synonymsForLanguage"
+      v-for="({ test_expression, visible_expression, globalIndex }, synonymIndex) in synonymsForLanguage"
       :key="synonymIndex"
       :message-key="`block/${block.uuid}/config/choices/${index}/text_tests/${globalIndex}/test_expression`">
       <template #input-control="{ isValid }">
         <expression-input
           ref="expressionInputs"
-          :current-expression="test_expression"
+          :current-expression="visible_expression"
```

Comment:
Display expressions, i.e., `@…` synonyms as is; while `@block.response` is displayed as plain text